samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -51,25 +52,39 @@ def prepare(params: dict) -> dict:
         information of the generated metadata files
     """
     output_dir_path = params.get("OutputDirPath")
+    terraform_application_dir = params.get("IACProjectPath", os.getcwd())
     if not output_dir_path:
         raise PrepareHookException("OutputDirPath was not supplied")
 
     try:
         # initialize terraform application
         LOG.info("Initializing Terraform application")
-        run(["terraform", "init"], check=True, capture_output=True)
+        run(["terraform", "init"], check=True, capture_output=True, cwd=terraform_application_dir)
```

Comment:
I made this comment in the last PR, but in light of UX I would like to ensure that we do have terraform installed, what kind of error message do we showcase if the terraform binary is not installed?

Reply:
We will have a separate task for this point. We are going also to have separate Exception for the following use cases:
 - Terraform is not installed
 -  Prepare Hook failed because of Terraform error (most likely means the tf application contains some issues)
 - Prepare Hook failed because any issue in our side.